Add `inverse-text` functional colour for white text on solid colour (#7178)
alphagov/govuk-frontend: CHANGELOG.md

### Recommended changes

> #### Use `inverse-text` instead of `govuk-colour("white")` when background is `brand`, `success` or `error`

Reviewer: ```suggestion
#### Use `inverse-text` instead of `govuk-colour("white")` for text on dark backgrounds
```

Just thinking we'll probably end up documenting the new colour something like this?

> We've introduced a new `inverse-text` functional colour to set the colour of the text when the background is another functional colour like `brand`, `success` or `error`.

Reviewer: ```suggestion
We've introduced a new `inverse-text` functional colour to use on dark backgrounds that use functional colours like `brand`, `success` or `error`.
```

If you were using `govuk-colour("white")` as text colour against such backgrounds, replace `govuk-colour("white")` with `govuk-functional-colour(inverse-text)`.

However, if the background was not a functional colour, keep using `govuk-colour("white")`.

> We made this change in [pull request #7178: Add `inverse-text` functional colour for white text on solid colour](https://github.com/alphagov/govuk-frontend/pull/7178)

Reviewer: Do we need to say anything about passing WCAG contrast with this colour or for guidance elsewhere?

Author: A little reminder probably won't hurt, maybe something like.
```suggestion
In both cases, make sure your text has a minimum contrast of 4.5:1 with the background to meet the ['Contrast (minimum)' criterion of the Web Content Accessibility Guidelines](https://www.w3.org/WAI/WCAG22/quickref/#contrast-minimum).

We made this change in [pull request #7178: Add `inverse-text` functional colour for white text on solid colour](https://github.com/alphagov/govuk-frontend/pull/7178)
```

### Recommended changes

> #### Use `inverse-text` instead of `govuk-colour("white")` for text on dark backgrounds

Reviewer 2: ```suggestion
#### Use `inverse-text` instead of `govuk-colour("white")` for text on dark backgrounds that use functional colours
```

> We've introduced a new `inverse-text` functional colour to use on dark backgrounds that use functional colours like `brand`, `success` or `error`.

Reviewer 2: ```suggestion
We've introduced a new `inverse-text` functional colour to use on dark backgrounds that use functional colours such as `brand`, `success` and `error`.
```

> If you were using `govuk-colour("white")` as text colour against such backgrounds, replace `govuk-colour("white")` with `govuk-functional-colour(inverse-text)`.

Reviewer 2: ```suggestion
If you're using `govuk-colour("white")` as a text colour on dark backgrounds that use functional colours, replace `govuk-colour("white")` with `govuk-functional-colour(inverse-text)`.
```

> However, if the background was not a functional colour, keep using `govuk-colour("white")`.

Reviewer 2: ```suggestion
However, if you're using a dark background that does not use a functional colour, keep using `govuk-colour("white")`.
```

In both cases, make sure your text has a minimum contrast of 4.5:1 with the background to meet the ['Contrast (minimum)' criterion of the Web Content Accessibility Guidelines](https://www.w3.org/WAI/WCAG22/quickref/#contrast-minimum).

> We made this change in [pull request #7178: Add `inverse-text` functional colour for white text on solid colour](https://github.com/alphagov/govuk-frontend/pull/7178)

Reviewer 2: ```suggestion
We made this change in [pull request #7178: Add `inverse-text` functional colour for white text on solid colour](https://github.com/alphagov/govuk-frontend/pull/7178).
```
